### ISSUE TITLE
Erronous halo size warning

### ISSUE THREAD

**Issue by rhodrin** [devitocodes/devito#1466]
Running e.g. `DEVITO_MPI=1 mpirun -np 4 python acoustic_example.py` results in the following warnings:
```
A space order of 4 and a halo size of Size(left=4, right=4) has been
                                 set but the current rank (2) has a domain size of
                                 only 65
A space order of 4 and a halo size of Size(left=4, right=4) has been
                                 set but the current rank (1) has a domain size of
                                 only 65
```
but this shouldn't be the case.

Note: no warning is produced for `-np 2`.

**Comment by rhodrin** [comment]
Just a note that this warning is the result of `SparseFunction`'s being under these criteria (which they shouldn't be). Fixed in #1380, so can close this after that's merged (which still may be a little while).

**Comment by rhodrin** [comment]
Scratch that, it turned out to be something else, but is fixed within that PR now anyway.

**Comment by rhodrin** [comment]
Owing to the planned changes to #1380 this is now fixed in #1493.
